packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -404,7 +416,16 @@ module.exports = {
       fileName: 'asset-manifest.json',
       publicPath: publicPath,
     }),
-  ],
+    // TypeScript type checking
+    paths.isTypeScript &&
+      new ForkTsCheckerWebpackPlugin({
+        async: false,
+        checkSyntacticErrors: true,
+        tsconfig: paths.appTSConfig,
+        tslint: paths.useTSLint ? paths.appTSLint : undefined,
```

Comment:
Remove support for TSLint for now please. People can run this separate if they really want it.